src/app/modules/form/public-form/public-form.controller.ts

```diff
@@ -451,6 +467,42 @@ export const handleFormAuthRedirect = [
   _handleFormAuthRedirect,
 ] as ControllerHandler[]
 
+/**
+ * NOTE: This is exported only for testing
+ * Logs user out of SP / CP By deleting cookie
+ * @param authType type of authentication
+ *
+ * @returns 200 with success message when user logs out successfully
+ * @returns 400 if authType is invalid
+ */
+export const _handleFormAuthLogout: ControllerHandler<
+  { authType: AuthType },
+  PublicFormAuthLogoutDto
+> = (req, res) => {
+  const { authType } = req.params
+
+  if (authType !== AuthType.SP && authType !== AuthType.CP) {
+    return res.status(400).json({ message: 'Invalid authType.' })
+  }
```

Comment:
the Joi validation already guarantees that the authType is SP or CP right? so we can type this as `{ authType: AuthType.SP | AuthType.CP }` and remove the unnecessary `if` statement

Reply:
edited